aiida/orm/utils/loaders.py

```diff
@@ -105,14 +105,20 @@ def _get_query_builder_id_identifier(cls, identifier, classes):
         :param classes: a tuple of orm classes to which the identifier should be mapped
         :returns: the query builder instance
         """
+        if project is None:
+            project = ['*']
+
+        if operator == 'like':
+            identifier = str(identifier) + '%'
+
         builder = QueryBuilder()
-        builder.append(cls=classes, tag='entity', project=['*'])
-        builder.add_filter('entity', {'id': identifier})
+        builder.append(cls=classes, tag='entity', project=project)
+        builder.add_filter('entity', {'id': {operator: identifier}})
 
         return builder
 
     @classmethod
-    def _get_query_builder_uuid_identifier(cls, identifier, classes, query_with_dashes):
+    def _get_query_builder_uuid_identifier(cls, identifier, classes, query_with_dashes, project=None):
```

Comment:
would you mind adding a docstring for `project` as well?

Reply:
Not at all, but since I realize there are some design decisions to be made and maybe in practice this is not necessarily very useful (see my post below), I first wanted to pass it by in this form